src/Microsoft.AspNetCore.Antiforgery/Internal/IAntiforgeryTokenGenerator.cs

```diff
@@ -20,9 +21,13 @@ public interface IAntiforgeryTokenGenerator
         /// Generates a request token corresponding to <paramref name="cookieToken"/>.
         /// </summary>
         /// <param name="httpContext">The <see cref="HttpContext"/> associated with the current request.</param>
+        /// <param name="principal">The claims-based principal to use for token generation.</param>
         /// <param name="cookieToken">A valid cookie token.</param>
         /// <returns>An <see cref="AntiforgeryToken"/>.</returns>
-        AntiforgeryToken GenerateRequestToken(HttpContext httpContext, AntiforgeryToken cookieToken);
+        AntiforgeryToken GenerateRequestToken(
+            HttpContext httpContext,
+            ClaimsPrincipal principal,
```

Comment:
Not strictly required, but it's nice to look at the code and see that `context.User` isn't used anywhere.